### ISSUE TITLE
Tile and Row custom style menus should be contained in the theme

### ISSUE THREAD

**Issue by djay** [plone/plone.app.mosaic#77]
Looks like this is already supported inside tinymce. Will it work with mosaic too?
https://github.com/plone/Products.CMFPlone/issues/492#issuecomment-100435844

**Comment by datakurre** [comment]
Technically yes, but how would we separate RichText styles, Tile styles and Row styles? TinyMCE's built-in support is only for RichtText styles. Also, not sure how much we can reuse TinyMCE API and how much we need to reimplement (in Python and in JS).

**Comment by datakurre** [comment]
Well, need to be bit more careful:

* Mosaic Editor is based on TinyMCE 4, but does not use its native menus or toolbars, which would benefit from TinyMCE's auto-import-feature. That's because each tile could have its own set of (tile type specific) formatting actions and probably that was not possible with TinyMCE 3 (used by the original Deco editor). Ramon has experimented with TinyMCE's native toolbar, but with that we may loose the ability to limit formatting options per tile type (or at least it would be re-implemented against TinyMCE's API).

* By default, TinyMCE styles are expected to be only add Rich Text formatting actions. Styles for tiles and rows would probably require their own files, because otherwise they would also appear in normal TinyMCE formatting menus, but then we could no longer benefit from that TinyMCE plugin, but would have to re-implement everything in Mosaic.

So, I probably was too optimistic. Yet, I agree that a styles sounds like a natural configuration method for styles :)

**Comment by datakurre** [comment]
Maybe p.a.mosaic could provide a theming plugin, which populates mosaic configuration during theme activation (and clean ups during deactivation). That would be the easiest way with the current editor configuration model.

**Comment by djay** [comment]
I think its important to make it consistent with how tinymce now does it in plone5 so we don’t have to teach users two ways to do things. ie it should parse a nominated css file and the manifest.

**Comment by datakurre** [comment]
@djay I don't think that TinyMCE code is re-usable, but the idea should work: a configuration line in theme manifest specifying the css file where the classes should be read from. Probably different files for tiles and rows would be the simpliest way. The existing tinymce-styles line should be used for rich text tile text styles.

(All this with assumption that no one has resources to rewrite the editor and we keep the current way to configure it.)

**Comment by djay** [comment]
+1
